README.md

```diff
@@ -7,620 +7,799 @@
+  {
+    path: "/",
+    element: <Home />
+  }, 
+  {
+    path: "/about",
+    element: <About />
+  },
+  {
+    path: "/login",
+    element: <Login />
+  },
+  {
+    path: "/profile/:id",
+    element: <UserProfile />
+  }
+])
+
+// ...render statements
+```
 
-function Login() {
+Notice that we added `:id` to the end of our `path` for our `UserProfile` route.
+This notation creates a `URL parameter` - a segment of our URL that can change
+and that contains data that we can use in our components.
+
+By including a URL parameter (or multiple parameters) in a route, we make that
+route _dynamic_ - this single route can actually have many different URLs! For
+example, the `/profile/1`, `/profile/2`, and `/profile/3` URLs will all lead to
+the same page. That page will just display different information depending on
+which URL is used!
+
+Let's update our `UserCard` component to start making use of our dynamic route:
+
+```jsx
+// UserCard.js
+import {Link} from "react-router-dom";
+
+function UserCard({id, name}) {
   return (
-    <div>
-      <h1>Login</h1>
-      <form>
-        <div>
-          <input type="text" name="username" placeholder="Username" />
-        </div>
-        <div>
-          <input type="password" name="password" placeholder="Password" />
-        </div>
-        <input type="submit" value="Submit" />
-      </form>
-    </div>
+    <article>
+        <h2>{name}</h2>
+        <p>
+          <Link to={`/profile/${id}`}>View profile</Link>
+        </p>
+    </article>
   );
+};
+
+export default UserCard;
+```
+
+We've used string interpolation to update the `to` prop of our `Link` component
+from `react-router-dom` to include the `id` of a user being rendered by a
+particular `UserCard` component. Now, when we click on one of these links, it
+will take us to the URL `/profile/<some-user-id>`, which will correspond with
+the `/profile/:id` route we set up in our router.
+
+Try it out! You should still see the `UserProfile` component being rendered as
+it was before, but the URl should show the `id` of whichever user you clicked
+on.
+
+Great! But our `UserProfile` component still isn't displaying specific user
+information.
+
+That's where the last piece of the puzzle comes into play - the `useParams`
+hook. `useParams` allows us to access the data we've stored in our URL
+parameters and use it within our components.
+
+Let's start by importing that into the top of our `UserProfile` component:
+`import { useParams } from 'react-router-dom'`.
+
+From there, we can invoke the hook to access the parameters we included in our
+URL route: `const params = useParams()`.
+
+If we `console.log` our new `params` variable, we'll see that it's an object.
+The keys in the object will be the parameters we defined in our route, and the
+values will be whatever we actually entered into our URL.
+
+If we click on `George Orwell's` card for example, our `params` object should
+look like this:
+
+```JavaScript
+{
+  id: "1"
 }
+```
 
```

Comment:
Great edit, thank you!

Reply:
This is a big improvement! I made a few changes to your suggestion - I said "component" instead of function, and "parameters" instead of "results". Let me know if you think those changes will cause problems.

Reply:
Nope, I think that's great!